XMonad/Hooks/DynamicLog.hs

```diff
@@ -746,6 +766,192 @@ xmobarStripTags tags = strip [] where
+statusBarHandleConfig' :: String       -- ^ The command line to launch the status bar
+                       -> X PP         -- ^ The pretty printing options
+                       -> IO StatusBarConfig
+statusBarHandleConfig' cmd xpp  = do
+    h <- spawnPipe cmd
+    return $ def { sbLogHook = xpp >>= \pp -> dynamicLogWithPP pp { ppOutput = hPutStrLn h } }
+
+-- $multiple
+-- A pattern that is often found in a lot of configs that want multiple status bars,
+-- generally goes something like this:
+--
+-- > main = do
+-- >   xmproc0 <- spawnPipe "xmobar -x 0 $HOME/.config/xmobar/xmobarrc_top"
+-- >   xmproc1 <- spawnPipe "xmobar -x 0 $HOME/.config/xmobar/xmobarrc_bottom"
+-- >   xmproc2 <- spawnPipe "xmobar -x 1 $HOME/.config/xmobar/xmobarrc1"
+-- >   xmonad $ def {
+-- >     ...
+-- >     , logHook = dynamicLogWithPP ppTop { ppOutput = hPutStrLn xmproc0 }
+-- >              >> dynamicLogWithPP ppBottom { ppOutput = hPutStrLn xmproc1 }
+-- >              >> dynamicLogWithPP pp1 { ppOutput = hPutStrLn xmproc2 }
+-- >     ...
+-- >   }
+--
+-- Which has a lot of boilerplate. Using the provided functions in this section, the amount of
+-- boilerplate is reduced.
+-- If you wish to have multiple status bars managed by xmonad, you can do so by creating
+-- the status bar configs and combining them with 'combineSBs':
+--
+-- > xmobarTop    = statusBarHandleConfig "xmobar -x 0 $HOME/.config/xmobar/xmobarrc_top"    ppTop
+-- > xmobarBottom = statusBarHandleConfig "xmobar -x 0 $HOME/.config/xmobar/xmobarrc_bottom" ppBottom
+-- > xmobar1      = statusBarHandleConfig "xmobar -x 1 $HOME/.config/xmobar/xmobarrc1"       pp1
+-- > main = xmonad =<< combineSBs [xmobarTop, xmobarBottom, xmobar1] myConfig
+--
+-- Or if you're feeling adventurous:
+--
+-- > myBars = map (uncurry statusBarHandleConfig) [ ("xmobar -x 0 $HOME/.config/xmobar/xmobarrc_top",    ppTop)
+-- >                                              , ("xmobar -x 0 $HOME/.config/xmobar/xmobarrc_bottom", ppBottom)
+-- >                                              , ("xmobar -x 1 $HOME/.config/xmobar/xmobarrc1",       pp1) ]
+-- > main = xmonad =<< combineSBs myBars myConfig
+--
+-- To change the default keybinding or action to toggle struts, you can do the following:
+--
+-- > ...
+-- > mySBCombinator = (withAction myAction . combineSBConfs <$> sequence myBars >>=). flip makeStatusBar
+-- > main = xmonad =<< mySBCombinator myConfig
+--
+-- if you also want to change the key, you can chain the transformations like so:
+--
+-- > ...
+-- > mySBCombinator = (withKey myToggleStrutsKey . withAction myAction . combineSBConfs <$> sequence myBars >>=)
+-- >     . flip makeStatusBar
+-- > ...
+--
+-- The above examples also work if the different status bars support different
+-- logging methods: you could do mix property logging and logging via standard input.
+-- One thing to keep in mind is that if you have multiple status bars with
+-- property logging, they must read from different properties.
```

Comment:
If they read from the same property, they will get the same content. It's not intuitive, but that's how `xmonadPropLog` works. I'll update this with @geekosaur 's comment

Reply:
Ah I see.  I do think it's intuitive that bars will display the same result if they read the same input, but your explanation made it sounds like something bad would happen if one tried that kind of setup :)  Updating this sounds like a good idea (especially since some people may want identical bars that are just on different monitors)

Reply:
Maybe my explanation was *slightly* influenced by me wondering for an hour why my bars were displaying the same thing